packages/flutter_tools/lib/src/version.dart

```diff
@@ -150,12 +152,14 @@ class FlutterVersion {
   String get engineRevisionShort => _shortGitRevision(engineRevision);
 
   void ensureVersionFile() {
-    globals.fs.file(globals.fs.path.join(Cache.flutterRoot, 'version')).writeAsStringSync(_frameworkVersion);
+    if (Cache.flutterRoot != null) {
```

Comment:
should we throw an exception if this is null? Seems like a bug if this wasn't initialized.

Reply:
From https://github.com/flutter/flutter/pull/79864/
> This is mostly the same API, with the caveat that flutterRoot now defaults to null. There were too many tests that failed when I made it late, though we should consider revisiting that in the future to catch cases where we're not running the setup we think we are

https://github.com/flutter/flutter/blob/0a59698ebe95eb02e47746e3c8dc2eb846372659/packages/flutter_tools/lib/src/cache.dart#L186-L187

I'm going to make this `Cache.flutterRoot!` instead so it crashes.

Reply:
SGTM